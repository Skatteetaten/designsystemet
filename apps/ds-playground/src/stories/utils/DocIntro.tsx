import { ReactNode } from 'react';

import { Link } from '@skatteetaten/ds-buttons';
import { Paragraph } from '@skatteetaten/ds-typography';

interface DocIntroProps {
  name: string;
  url: string;
}

export const DocIntro = ({ name, url }: DocIntroProps): ReactNode => {
  return (
    <Paragraph variant={'ingress'}>
      {`Se `}
      <Link href={url}>{name}</Link>
      {` p√• stil og tone for eksempler og
        retningslinjer for denne komponenten.`}
    </Paragraph>
  );
};
