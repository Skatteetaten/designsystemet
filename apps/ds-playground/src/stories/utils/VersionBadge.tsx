import { ReactNode } from 'react';

interface VersionBadgeProps {
  packageName: string;
}

export const VersionBadge = ({ packageName }: VersionBadgeProps): ReactNode => {
  let color = 'green';

  let version =
    import.meta.env.DS_VERSIONS[`${packageName.replaceAll('-', '_')}`] ?? '..';
  if (import.meta.env.MODE === 'development') {
    color = 'red';
    version = 'developing';
  }
  if (import.meta.env['STORYBOOK_BUILD_VERSION']?.includes('SNAPSHOT')) {
    color = 'blue';
  }

  const normalizedPackageName = packageName.replaceAll('-', '--');
  const normalizedVersion = version.replaceAll('-', '--').replaceAll('_', '__');

  return (
    <img
      src={`https://img.shields.io/badge/%40skatteetaten%2Fds--${normalizedPackageName}-${normalizedVersion}-${color}?logo=npm`}
      alt={`@skatteetaten/ds-${packageName}@${version}`}
    />
  );
};
