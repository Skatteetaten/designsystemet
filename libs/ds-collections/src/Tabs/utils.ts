export const valueRegex = /^[A-Za-z][\w\-.:]*$/;
