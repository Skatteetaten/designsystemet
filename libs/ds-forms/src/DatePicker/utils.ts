import { format, isValid, parse } from 'date-fns';

export const formatDateForInput = (
  dateFormat: string,
  date?: Date | null
): string => {
  return !isValid(date) ? '' : format(date as Date, dateFormat);
};

const COMMON_INPUT_FORMATS = [
  'dd.MM.yy',
  'dd/MM/yy',
  'dd-MM-yy',
  'dd.MM.yyyy',
  'dd/MM/yyyy',
  'dd-MM-yyyy',
  'ddMM',
  'ddMMyy',
  'ddMMyyyy',
];

export const parseDateFromInput = (
  value: string,
  dateFormat: string
): Date | null => {
  const dateFormats = Array.from(
    new Set([...COMMON_INPUT_FORMATS, dateFormat])
  );

  for (const format of dateFormats) {
    const date = parse(value, format, new Date());
    if (isValid(date)) {
      return date;
    }
  }

  return null;
};
