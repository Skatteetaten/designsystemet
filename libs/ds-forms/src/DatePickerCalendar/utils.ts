import { dsI18n } from '@skatteetaten/ds-core-utils';
import {
  addDays,
  getWeekOfMonth,
  isAfter,
  isBefore,
  isSunday,
  isToday,
} from 'date-fns';

const lastValidYear = 9999;
const sunday = 0;

function getDaysFromPrevMonth(datesInSelectedMonth: Date[]): number {
  const firstDayInSelectedMonth = datesInSelectedMonth[0].getDay();
  let daysFromPrevMonthInFirstWeekOfSelectedMonth;
  if (firstDayInSelectedMonth === sunday) {
    daysFromPrevMonthInFirstWeekOfSelectedMonth = 6;
  } else {
    daysFromPrevMonthInFirstWeekOfSelectedMonth = firstDayInSelectedMonth - 1;
  }

  return daysFromPrevMonthInFirstWeekOfSelectedMonth;
}

function getDaysFromNextMonth(datesInSelectedMonth: Date[]): number {
  const lastDayInSelectedMonth =
    datesInSelectedMonth[datesInSelectedMonth.length - 1].getDay();
  let daysFromNextMonthInLastWeekOfSelectedMonth;
  if (lastDayInSelectedMonth === sunday) {
    daysFromNextMonthInLastWeekOfSelectedMonth = 0;
  } else {
    daysFromNextMonthInLastWeekOfSelectedMonth = 7 - lastDayInSelectedMonth;
  }

  return daysFromNextMonthInLastWeekOfSelectedMonth;
}

function getLastDateInPrevMonth(year: number, monthIndex: number): Date {
  const lastDateInPrevMonth = new Date(year, monthIndex, 0);
  lastDateInPrevMonth.setFullYear(lastDateInPrevMonth.getFullYear());

  return lastDateInPrevMonth;
}

function getFirstDateInNextMonth(year: number, monthIndex: number): Date {
  const firstDateInNextMonth = new Date(year, monthIndex + 1, 1);
  firstDateInNextMonth.setFullYear(firstDateInNextMonth.getFullYear());

  return firstDateInNextMonth;
}

interface Cell {
  date: Date;
  isAdjacentMonth: boolean;
  isToday: boolean;
  text: string;
  disabled: boolean;
}

function getCalendarCells(
  year: number,
  monthIndex: number,
  minDate?: Date,
  maxDate?: Date,
  disabledDates?: Date[]
): Cell[] {
  const dates = getDatesInMonth(year, monthIndex);
  const cells: Cell[] = [];

  const daysFromPrevMonth = getDaysFromPrevMonth(dates);
  const lastDateInPrevMonth = getLastDateInPrevMonth(year, monthIndex);

  for (let i = 0; i < daysFromPrevMonth; i++) {
    const date = new Date(lastDateInPrevMonth);

    const cell: Cell = {
      date,
      isAdjacentMonth: true,
      disabled:
        disabledDates?.some((d) => d.getTime() === date.getTime()) ||
        isNotInAllowedRange(date, minDate, maxDate),
      isToday: isToday(date),
      text: String(date.getDate()),
    };

    cells.unshift(cell);

    lastDateInPrevMonth.setDate(lastDateInPrevMonth.getDate() - 1);
  }

  dates.forEach((date) => {
    const cell: Cell = {
      date,
      isAdjacentMonth: false,
      disabled:
        disabledDates?.some((d) => d.getTime() === date.getTime()) ||
        isNotInAllowedRange(date, minDate, maxDate),
      isToday: isToday(date),
      text: String(date.getDate()),
    };

    cells.push(cell);
  });

  const daysFromNextMonth = getDaysFromNextMonth(dates);
  const firstDateInNextMonth = getFirstDateInNextMonth(year, monthIndex);
  for (let i = 0; i < daysFromNextMonth; i++) {
    const date = new Date(firstDateInNextMonth);

    const cell: Cell = {
      date,
      isAdjacentMonth: true,
      disabled:
        disabledDates?.some((d) => d.getTime() === date.getTime()) ||
        isNotInAllowedRange(date, minDate, maxDate),
      isToday: isToday(date),
      text: String(date.getDate()),
    };

    cells.push(cell);

    firstDateInNextMonth.setDate(firstDateInNextMonth.getDate() + 1);
  }

  return cells;
}

function getDatesInMonth(year: number, monthIndex: number): Date[] {
  const date = new Date(year, monthIndex);
  date.setFullYear(year);
  date.setDate(1);

  const month = date.getMonth();
  const dates: Date[] = [];
  while (date.getMonth() === month) {
    dates.push(new Date(date));
    date.setDate(date.getDate() + 1);
    date.setHours(0, 0, 0);
  }
  return dates;
}

export function getCalendarRows(
  year: number | string,
  monthIndex: number,
  minDate?: Date,
  maxDate?: Date,
  disabledDates?: Date[]
): Array<Cell>[] {
  year = findValidYear(year);
  const cells = getCalendarCells(
    year,
    monthIndex,
    minDate,
    maxDate,
    disabledDates
  );
  const rows: Array<Cell>[] = [];
  for (let i = 0; i < cells.length; i += 7) {
    rows.push(cells.slice(i, i + 7));
  }

  return rows;
}

export const isNotInAllowedRange = (
  date: Date,
  minDate?: Date,
  maxDate?: Date
): boolean => {
  minDate && minDate.setHours(0, 0, 0);
  return (
    (minDate ? isBefore(date, minDate) : false) ||
    (maxDate ? isAfter(date, maxDate) : false) ||
    date.getFullYear() > lastValidYear
  );
};

export const getNameOfMonthsAndDays = (): {
  monthNames: string[];
  dayNames: string[];
} => {
  return {
    monthNames: [
      dsI18n.t('ds_forms:datepicker.January'),
      dsI18n.t('ds_forms:datepicker.February'),
      dsI18n.t('ds_forms:datepicker.March'),
      dsI18n.t('ds_forms:datepicker.April'),
      dsI18n.t('ds_forms:datepicker.May'),
      dsI18n.t('ds_forms:datepicker.June'),
      dsI18n.t('ds_forms:datepicker.July'),
      dsI18n.t('ds_forms:datepicker.August'),
      dsI18n.t('ds_forms:datepicker.September'),
      dsI18n.t('ds_forms:datepicker.October'),
      dsI18n.t('ds_forms:datepicker.November'),
      dsI18n.t('ds_forms:datepicker.December'),
    ],
    dayNames: [
      dsI18n.t('ds_forms:datepicker.Mon'),
      dsI18n.t('ds_forms:datepicker.Tue'),
      dsI18n.t('ds_forms:datepicker.Wed'),
      dsI18n.t('ds_forms:datepicker.Thu'),
      dsI18n.t('ds_forms:datepicker.Fri'),
      dsI18n.t('ds_forms:datepicker.Sat'),
      dsI18n.t('ds_forms:datepicker.Sun'),
    ],
  };
};

export function findValidYear(year: string | number): number {
  return year === '' || year === 0 ? new Date().getFullYear() : Number(year);
}

export function getGridIdxForDate(focusableDate: Date): string {
  const colIdx = isSunday(focusableDate) ? 6 : focusableDate.getDay() - 1;
  const rowIdx = getWeekOfMonth(focusableDate, { weekStartsOn: 1 }) - 1;

  return `${rowIdx}${colIdx}`;
}

export const getFirstFocusableDate = (
  selectedDate: Date,
  minDate?: Date,
  maxDate?: Date
): Date => {
  if (isNotInAllowedRange(selectedDate, minDate, maxDate)) {
    let focusableDate = undefined;
    if (maxDate && isAfter(selectedDate, maxDate)) {
      focusableDate = maxDate;
    } else if (minDate && isBefore(selectedDate, minDate)) {
      focusableDate = minDate;
    }

    if (
      focusableDate &&
      !isNotInAllowedRange(focusableDate, minDate, maxDate)
    ) {
      return focusableDate;
    }
  }

  return selectedDate;
};

export const findNextAvailableDate = (
  startDate: Date,
  disabledDates?: Date[]
): Date => {
  if (!disabledDates || disabledDates.length === 0) {
    return addDays(startDate, 1);
  }

  const maxDate = new Date().setFullYear(lastValidYear);
  const disabledTimestamps = new Set(
    disabledDates.map((date) => date.getTime())
  );

  let currentDate = addDays(startDate, 1);
  let currentTimestamp = currentDate.getTime();

  while (
    disabledTimestamps.has(currentTimestamp) &&
    isBefore(currentDate, maxDate)
  ) {
    currentDate = addDays(currentDate, 1);
    currentTimestamp = currentDate.getTime();
  }

  return currentDate;
};

export const findPreviousAvailableDate = (
  startDate: Date,
  disabledDates?: Date[]
): Date => {
  if (!disabledDates || disabledDates.length === 0) {
    return addDays(startDate, -1);
  }

  const minDate = new Date('1000-01-01');
  const disabledTimestamps = new Set(
    disabledDates.map((date) => date.getTime())
  );

  let currentDate = addDays(startDate, -1);
  let currentTimestamp = currentDate.getTime();

  while (
    disabledTimestamps.has(currentTimestamp) &&
    isBefore(minDate, currentDate)
  ) {
    currentDate = addDays(currentDate, -1);
    currentTimestamp = currentDate.getTime();
  }

  return currentDate;
};
