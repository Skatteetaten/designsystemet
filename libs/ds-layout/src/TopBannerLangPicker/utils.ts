import { Languages } from '@skatteetaten/ds-core-utils';

import { TopBannerLangPickerProps } from './TopBannerLangPicker.types';

export const convertLocaleToLang = (locale: Languages): string => {
  return locale.substring(0, locale.indexOf('_'));
};

export type LanguageItem = {
  lang: string;
  displayName: string;
};

export type LanguageItems = {
  [key: string]: LanguageItem;
};

export const isLanguages = (value: unknown): value is Languages => {
  return Object.values(Languages).includes(value as Languages);
};

export const getCurrentLanguages = (
  includeSami: boolean,
  additionalLanguages: TopBannerLangPickerProps['additionalLanguages'] = []
): LanguageItems => {
  const defaultLanguages: LanguageItems = {
    nb: {
      lang: 'nb',
      displayName: 'Bokmål',
    },
    nn: {
      lang: 'nn',
      displayName: 'Nynorsk',
    },
    en: {
      lang: 'en',
      displayName: 'English',
    },
  };
  const se = {
    se: {
      lang: 'se',
      displayName: 'Sámegiella',
    },
  };

  const additionalLanguagesAsObject =
    additionalLanguages?.reduce<LanguageItems>((acc, curr) => {
      acc[curr.lang] = curr;
      return acc;
    }, {});

  return includeSami
    ? { ...defaultLanguages, ...se, ...additionalLanguagesAsObject }
    : { ...defaultLanguages, ...additionalLanguagesAsObject };
};
