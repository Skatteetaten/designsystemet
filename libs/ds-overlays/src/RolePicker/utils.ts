import { formatter } from '@skatteetaten/ds-core-utils';

export const formatOrganisationNumber = (
  organisationNumber: string
): string => {
  const result = formatter({
    type: 'organisationNumber',
    value: organisationNumber,
  });

  return result.value;
};

export const formatNationalIdentityNumber = (
  nationalIdentityNumber: string
): string => {
  const result = formatter({
    type: 'nationalIdentityNumber',
    value: nationalIdentityNumber,
  });

  return result.value;
};
