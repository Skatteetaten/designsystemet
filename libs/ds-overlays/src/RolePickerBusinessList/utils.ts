import { dsI18n } from '@skatteetaten/ds-core-utils';

import { Business } from '../RolePicker/RolePicker.types';

const unitTypesToHide = [
  'AS',
  'DA',
  'ANS',
  'ASA',
  'SE',
  'BA',
  'SA',
  'SCE',
  'STI',
  'IKS',
  'KF',
  'FKF',
  'RHF',
  'HF',
  'NUF',
];

export const getBusinessTitle = (business: Business): string => {
  let title = business.name;
  if (!unitTypesToHide.includes(business.unitType)) {
    title += ` ${business.unitType}`;
  }

  if (business.isDeleted) {
    title += ` (${dsI18n.t('ds_overlays:rolepicker.Deleted')})`;
  }

  return title;
};
