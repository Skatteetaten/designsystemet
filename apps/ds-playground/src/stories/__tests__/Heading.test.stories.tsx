import { headingAsArr } from '@skatteetaten/ds-core-utils';
import {
  Heading,
  HeadingProps,
  headingLevelArr,
} from '@skatteetaten/ds-typography';
import { StoryFn, Meta, StoryObj } from '@storybook/react';
import { expect, within } from '@storybook/test';

import { wrapper } from './testUtils/storybook.testing.utils';

const meta = {
  component: Heading,
  title: 'Tester/Heading',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    canBeManuallyFocused: { table: { disable: true } },
    children: {
      table: { disable: true },
      control: 'text',
    },
    hasSpacing: { table: { disable: true } },
    as: {
      table: { disable: true },
    },
    level: {
      table: { disable: true },
    },
  },
} satisfies Meta<typeof Heading>;
export default meta;
type Story = StoryObj<typeof meta>;

const headingAsTest = 'h2';
const defaultArgs: HeadingProps = {
  as: headingAsTest,
  level: 2,
  children: 'Dette er en heading',
};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLHeadingElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: { imageSnapshot: { disable: true } },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const heading = canvas.getByRole('heading');
    await expect(heading).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5)',
  args: {
    ...defaultArgs,
    id: 'htmlId',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const heading = canvas.getByRole('heading');
    await expect(heading).toHaveClass('dummyClassname');
    await expect(heading).toHaveAttribute('id', 'htmlId');
    await expect(heading).toHaveAttribute('lang', 'nb');
    await expect(heading).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

const TemplateWithAllLevels: StoryFn<typeof Heading> = (args) => (
  <>
    {headingLevelArr.map((headingLevel, item) => {
      return (
        <Heading key={`level_${item}`} {...args} level={headingLevel}>
          {`Heading level ${headingLevel}`}
        </Heading>
      );
    })}
  </>
);

export const Defaults = {
  render: TemplateWithAllLevels,
  name: 'Defaults All Levels (A1, A2)',
  args: {
    ...defaultArgs,
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const headings = canvas.getAllByRole('heading', { level: 2 });
    for (const [index, heading] of headings.entries()) {
      await expect(heading).toBeInTheDocument();
      await expect(heading.classList.toString()).toContain(
        `heading_level${headingLevelArr[index]}`
      );
    }
  },
} satisfies Story;

export const DefaultsMobile = {
  render: TemplateWithAllLevels,
  name: 'Defaults All Levels On Small Screen (A1, A2)',
  args: {
    ...defaultArgs,
  },
  parameters: {
    viewport: {
      defaultViewport: '--breakpoint-xs',
    },
  },
} satisfies Story;

export const LevelsWithSpacing = {
  render: TemplateWithAllLevels,
  name: 'With Spacing All Levels (A4)',
  args: {
    ...defaultArgs,
    hasSpacing: true,
  },
  argTypes: {
    hasSpacing: {
      table: { disable: false },
    },
  },
} satisfies Story;

export const LevelsWithSpacingMobile = {
  render: TemplateWithAllLevels,
  name: 'With Spacing All Levels On Small Screen (A4)',
  args: {
    ...defaultArgs,
    hasSpacing: true,
  },
  argTypes: {
    hasSpacing: {
      table: { disable: false },
    },
  },
  parameters: {
    viewport: {
      defaultViewport: '--breakpoint-xs',
    },
  },
} satisfies Story;

const AsTemplate: StoryFn<typeof Heading> = (args) => (
  <>
    {headingAsArr.map((heading) => (
      <Heading key={`heading-${heading}`} {...args} as={heading}>
        {`Heading ${heading}`}
      </Heading>
    ))}
  </>
);

export const WithAs = {
  render: AsTemplate,
  name: 'With As (B1)',
  args: {
    ...defaultArgs,
  },
  parameters: {
    viewport: {
      viewPortHeight: 1200,
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const headings = canvas.getAllByRole('heading');
    for (const [index, heading] of headings.entries()) {
      await expect(heading.tagName).toBe(
        headingAsArr[index].toLocaleUpperCase()
      );
    }
  },
} satisfies Story;

const TemplateWithMarkup: StoryFn<typeof Heading> = (args) => (
  <>
    <Heading {...args} hasSpacing>
      {'Dette er den '}
      <a href={'https://www.skatteetaten.no'}>{'fyneste markup headingen'}</a>
      <em>{' som finnes i italic '}</em>
      <strong>{'og strong '}</strong>
      <mark>{'mark '}</mark>
      <code>{'code '}</code>
      {'med spacing under og en skrivefeil'}
    </Heading>
    <Heading {...args} />
  </>
);

export const WithMarkup = {
  render: TemplateWithMarkup,
  name: 'With Markup And String (A3, B2)',
  args: {
    ...defaultArgs,
    children: 'Dette er den fineste string headingen uten markup',
  },
  argTypes: {
    children: {
      table: { disable: false },
      control: { disable: true },
    },
  },
  parameters: {
    imageSnapshot: {
      hover: `${wrapper} > h2 > a`,
      focus: `${wrapper} > h2 > a`,
    },
  },
} satisfies Story;

export const WithCanBeManuallyFocused: Story = {
  render: (args) => <Heading {...args} />,
  args: {
    ...defaultArgs,
    canBeManuallyFocused: true,
  },
  argTypes: {
    canBeManuallyFocused: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const heading = canvas.getByRole('heading');
    heading.focus();
    await expect(heading).toBeInTheDocument();
    await expect(heading).toHaveAttribute('tabIndex', '-1');
  },
} satisfies Story;
