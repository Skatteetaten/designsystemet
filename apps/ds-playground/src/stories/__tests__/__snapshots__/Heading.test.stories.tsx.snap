// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Heading Defaults play-test 1`] = `
<div data-test-block="true">
  <h2 class="_heading_ppf3o_1 _heading_level1_ppf3o_14  ">
    Heading level 1
  </h2>
  <h2 class="_heading_ppf3o_1 _heading_level2_ppf3o_23  ">
    Heading level 2
  </h2>
  <h2 class="_heading_ppf3o_1 _heading_level3_ppf3o_32  ">
    Heading level 3
  </h2>
  <h2 class="_heading_ppf3o_1 _heading_level4_ppf3o_41  ">
    Heading level 4
  </h2>
  <h2 class="_heading_ppf3o_1 _heading_level5_ppf3o_45  ">
    Heading level 5
  </h2>
</div>
`;

exports[`Tester/Heading DefaultsMobile smoke-test 1`] = `
<div data-test-block="true">
  <h2 class="_heading_ppf3o_1 _heading_level1_ppf3o_14  ">
    Heading level 1
  </h2>
  <h2 class="_heading_ppf3o_1 _heading_level2_ppf3o_23  ">
    Heading level 2
  </h2>
  <h2 class="_heading_ppf3o_1 _heading_level3_ppf3o_32  ">
    Heading level 3
  </h2>
  <h2 class="_heading_ppf3o_1 _heading_level4_ppf3o_41  ">
    Heading level 4
  </h2>
  <h2 class="_heading_ppf3o_1 _heading_level5_ppf3o_45  ">
    Heading level 5
  </h2>
</div>
`;

exports[`Tester/Heading LevelsWithSpacing smoke-test 1`] = `
<div data-test-block="true">
  <h2 class="_heading_ppf3o_1 _heading_level1_ppf3o_14 _heading_hasSpacingMedium_ppf3o_55 ">
    Heading level 1
  </h2>
  <h2 class="_heading_ppf3o_1 _heading_level2_ppf3o_23 _heading_hasSpacingSmallMobileMediumOtherwise_ppf3o_58 ">
    Heading level 2
  </h2>
  <h2 class="_heading_ppf3o_1 _heading_level3_ppf3o_32 _heading_hasSpacingSmall_ppf3o_52 ">
    Heading level 3
  </h2>
  <h2 class="_heading_ppf3o_1 _heading_level4_ppf3o_41 _heading_hasSpacingSmall_ppf3o_52 ">
    Heading level 4
  </h2>
  <h2 class="_heading_ppf3o_1 _heading_level5_ppf3o_45 _heading_hasSpacingExtraSmall_ppf3o_49 ">
    Heading level 5
  </h2>
</div>
`;

exports[`Tester/Heading LevelsWithSpacingMobile smoke-test 1`] = `
<div data-test-block="true">
  <h2 class="_heading_ppf3o_1 _heading_level1_ppf3o_14 _heading_hasSpacingMedium_ppf3o_55 ">
    Heading level 1
  </h2>
  <h2 class="_heading_ppf3o_1 _heading_level2_ppf3o_23 _heading_hasSpacingSmallMobileMediumOtherwise_ppf3o_58 ">
    Heading level 2
  </h2>
  <h2 class="_heading_ppf3o_1 _heading_level3_ppf3o_32 _heading_hasSpacingSmall_ppf3o_52 ">
    Heading level 3
  </h2>
  <h2 class="_heading_ppf3o_1 _heading_level4_ppf3o_41 _heading_hasSpacingSmall_ppf3o_52 ">
    Heading level 4
  </h2>
  <h2 class="_heading_ppf3o_1 _heading_level5_ppf3o_45 _heading_hasSpacingExtraSmall_ppf3o_49 ">
    Heading level 5
  </h2>
</div>
`;

exports[`Tester/Heading WithAs play-test 1`] = `
<div data-test-block="true">
  <h1 class="_heading_ppf3o_1 _heading_level2_ppf3o_23  ">
    Heading h1
  </h1>
  <h2 class="_heading_ppf3o_1 _heading_level2_ppf3o_23  ">
    Heading h2
  </h2>
  <h3 class="_heading_ppf3o_1 _heading_level2_ppf3o_23  ">
    Heading h3
  </h3>
  <h4 class="_heading_ppf3o_1 _heading_level2_ppf3o_23  ">
    Heading h4
  </h4>
  <h5 class="_heading_ppf3o_1 _heading_level2_ppf3o_23  ">
    Heading h5
  </h5>
  <h6 class="_heading_ppf3o_1 _heading_level2_ppf3o_23  ">
    Heading h6
  </h6>
</div>
`;

exports[`Tester/Heading WithAttributes play-test 1`] = `
<div data-test-block="true">
  <h2 id="htmlId"
      class="_heading_ppf3o_1 _heading_level2_ppf3o_23  dummyClassname"
      lang="nb"
      data-testid="123ID"
  >
    Dette er en heading
  </h2>
</div>
`;

exports[`Tester/Heading WithMarkup smoke-test 1`] = `
<div data-test-block="true">
  <h2 class="_heading_ppf3o_1 _heading_level2_ppf3o_23 _heading_hasSpacingSmallMobileMediumOtherwise_ppf3o_58 ">
    Dette er den
    <a href="https://skatteetaten.no">
      fyneste markup headingen
    </a>
    <em>
      som finnes i italic
    </em>
    <strong>
      og strong
    </strong>
    <mark>
      mark
    </mark>
    <code>
      code
    </code>
    med spacing under og en skrivefeil
  </h2>
  <h2 class="_heading_ppf3o_1 _heading_level2_ppf3o_23  ">
    Dette er den fineste string headingen uten markup
  </h2>
</div>
`;

exports[`Tester/Heading WithRef play-test 1`] = `
<div data-test-block="true">
  <h2 class="_heading_ppf3o_1 _heading_level2_ppf3o_23  "
      id="dummyIdForwardedFromRef"
  >
    Dette er en heading
  </h2>
</div>
`;
